test/core/util/port_isolated_runtime_environment.cc

```diff
@@ -28,18 +29,21 @@
 #include "test/core/util/port.h"
 
 #define MIN_PORT 49152
-#define MAX_PORT 65536
+#define MAX_PORT 65535
 
 int get_random_starting_port() {
   srand(gpr_now(GPR_CLOCK_REALTIME).tv_nsec);
-  return rand() % (MAX_PORT - MIN_PORT + 1) + MIN_PORT;
+  double rnd = static_cast<double>(rand()) /
+               (static_cast<double>(RAND_MAX) + 1.0);  // values from [0,1)
+  return static_cast<int>(rnd * (MAX_PORT - MIN_PORT + 1)) + MIN_PORT;
```

Comment:
wouldn't this still be able to return 65536 if rnd is very close to 1?

Reply:
double has enough precision to represent RAND_MAX / (RAND_MAX + 1.0)  as distinct from 1  (I tried).  2147483647 / (2147483647 +1) = 0.99999999953